### ISSUE TITLE
Make Check Fails Because libarb.so.0 is not found

### ISSUE THREAD

**Issue by rickyefarr** [fredrik-johansson/arb#51]
With the newest changes to the configure file, the shared object file libarb.so.0.0.0 is created.  When running the tests, it results in failure due to "libarb.so.0" not found.

Also, when using the --prefix= flag with ./configure, the include files are placed in the ./include file (as expected) but the library files are now placed above the lib directory.

**Comment by rickyefarr** [comment]
Probably related is the following error, but I'm not sure how:

sin_cos....FAIL: containment (sin)

a = (-140737488355327 * 2^-2) +/- (536874924 * 2^-288)

b = (-20274238756137371334080495 * 2^-85) +/- (541065218 * 2^-114)

../Makefile.subdirs:84: recipe for target '../build/arb/test/t-sin_cos_RUN' failed
make[1]: *** [../build/arb/test/t-sin_cos_RUN] Aborted (core dumped)
make[1]: Leaving directory '/home/therickaman/usr/local/arb/arb'
Makefile:168: recipe for target 'check' failed

**Comment by fredrik-johansson** [comment]
I've rolled back the configure changes temporarily.

**Comment by fredrik-johansson** [comment]
You reported the sin_cos failure before; did it just come back, or have you had it continuously?

**Comment by SnarkBoojum** [comment]
Make check ran perfectly here. Did you start from clean sources?

**Comment by fredrik-johansson** [comment]
I got the same problem (libarb.so.0 not found) after reconfiguring.

**Comment by SnarkBoojum** [comment]
Wait, I might have checked the wrong directory... sigh.

**Comment by rickyefarr** [comment]
I just updated from Ubuntu 14.04 to Ubuntu 15.04 for the intel graphics drivers.  In the process, gcc was updated.  The sin_cos issue was not a problem with compiling with gcc before.  When compiling arb with icc, the sin_cos issue arose (before the update).  Now, I can't get rid of the issue.  I've changed the CPATH, C_INCLUDE_PATH, LD_LIBRARY_PATH, and all the other environment variables I can think of to fix the sin_cos issue.

**Comment by fredrik-johansson** [comment]
Compiling with icc is not really supported since it does unsafe floating-point optimizations by default. I'm not sure if this is the cause here, but you could perhaps try compiling with -fp-model precise or -fp-model source and see if that makes a difference.

Otherwise, it would be helpful to have all the details: can you print out the exact values of iter, prec, q and t where it fails?

**Comment by rickyefarr** [comment]
I'm compiling with gcc, I'm just not sure if somehow the MLK library is being used for the math library.  I'll get those values as soon as I can.

**Comment by fredrik-johansson** [comment]
I hope the latest soname patch works correctly.

**Comment by SnarkBoojum** [comment]
So, what is the status?

**Comment by rickyefarr** [comment]
This seems to work for me.  Still have issues with the sin-cos, but it appears to be a separate issue from the title of this post.  As far as the issue in the title, I was able to install Arb.  I think that the sine-cosine issue mentioned above is a separate issue.  So, for that, it appears to be fixed.

With regard to the sine-cosine issue, unfortunately, I can not test this at the moment.  My motherboard fried, and I believe that installing icc somehow caused gcc to link the C math header, with the MLK math library.  I'll know more probably mid-week next week when a new motherboard is installed in my computer.  I'll report back then.

One final note, if my memory serves me correctly,  the sine-cosine function is the only function that fails under compilation with ICC (with gmp, flint, and mpfr installed with ICC as well with the MLK library) and this is without the -fp-precise flag.  I *think* I commented out that one test, and it passed all others.  If this is the case, it would be interesting to track down why there is a failure.  I'll report back about this when I get a new motherboard next week.
